### PR TITLE
fix: empty env variable

### LINKED ISSUE #11455: Empty string environment variable value treated as missing
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

If an environment variable is defined but is empty, Serverless reports it as 'Value not found at "env" source'.

The value _is_ found, it's just that value happens to be an empty string.

If in my .env file I have

```env
FOO=
```
How the environment variable is populated is irrelevant, the issue is in the framework code that checks it.

Logically, this should work. `fooValue` should get the value of `''` (empty string).
```yml
custom:
  fooValue: ${env:FOO} 
```

Instead this happens:

```
Cannot resolve serverless.yml: Variables resolution errored with:
  - Cannot resolve variable at "custom.fooValue": Value not found at "env" source
```

This following works but is strange:  an empty string is used as a default value if the value happens to be an empty string 🤯 

```yml
custom:
  fooValue: ${env:FOO, ''} 
```

The problem is this line: https://github.com/serverless/serverless/blob/main/lib/configuration/variables/sources/env.js#L24

Instead of 

```js
if (!process.env[address])
```

it should be 

```js
if (typeof process.env[address] === 'undefined')
```

### Service configuration (serverless.yml) content

```yaml
service: fooTest

custom:
  fooValue: ${env:FOO} 

provider:
  name: aws
```

### Command name and used flags

FOO= npx sls package  

### Command output

```shell
Environment: darwin, node 16.15.1, framework 3.23.0 (local), plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Cannot resolve serverless.yml: Variables resolution errored with:
  - Cannot resolve variable at "custom.fooValue": Value not found at "env" source
```

### Environment information

```shell
Framework Core: 3.23.0 (local)
Plugin: 6.2.2
SDK: 4.3.2
```

### PR DESCRIPTION
**Pull request by alanneves**
closes #11455

### CODE REVIEW

File: lib/configuration/variables/sources/env.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ module.exports = {
       errorCode: 'INVALID_ENV_SOURCE_ADDRESS',
     });
 
-    if (!process.env[address]) missingEnvVariables.add(address);
+    if (typeof process.env[address] === 'undefined') missingEnvVariables.add(address);
     return { value: process.env[address] || null, isPending: !isSourceFulfilled };
```
Comment (medikoo) [inline lib/configuration/variables/sources/env.js:25]: Let's use `process.env[address] == null` check instead. Also we need to update it below so `''` is not overridden with null

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @alanneves! Please see my comment

File: lib/configuration/variables/sources/env.js (hunk 1 above)
Comment (alanneves) [reply lib/configuration/variables/sources/env.js]: > Let's use `process.env[address] == null` check instead. Also we need to update it below so `''` is not overridden with null

Suggestions applied, is that correct?

File: lib/configuration/variables/sources/env.js [hunk 2]
```diff
@@ -21,8 +21,8 @@ module.exports = {
       errorCode: 'INVALID_ENV_SOURCE_ADDRESS',
     });
 
-    if (!process.env[address]) missingEnvVariables.add(address);
-    return { value: process.env[address] || null, isPending: !isSourceFulfilled };
+    if (process.env[address] == null) missingEnvVariables.add(address);
+    return { value: process.env[address], isPending: !isSourceFulfilled };
```
Comment (medikoo) [inline lib/configuration/variables/sources/env.js:25]: It should be: `value: process.env[address] == null ? null : process.env[address]`

Comment (alanneves) [reply lib/configuration/variables/sources/env.js]: @medikoo fixed

**Review by medikoo — CHANGES_REQUESTED** [review]
@alanneves change looks good now. There's just style issue, as introduced changes do not follow Prettier formatting.

You can easily fix it locally by running `npm run prettify:updated`

**Comment by alanneves** [comment]
@medikoo I applied the fix in the code style!

**Review by medikoo — APPROVED** [review]
Thank you!

**Review by medikoo — COMMENTED** [review]
@alanneves Sorry, just one more thing.

Can we add here: https://github.com/serverless/serverless/blob/294e45e99f29c8a18b91085ed473557925b345eb/test/unit/lib/configuration/variables/sources/env.test.js#L16-L20 a test case that confirms that empty string  env var is resolved without issues?

**Comment by akshatakulkarni25** [comment]
Hey, I am interested in writing the test case. Could you please help me understand the issue.

**Comment by maksym-opanasenko** [comment]
Hi guys, is it possible to merge the PR?
It's quite annoying to go to the node_modules and edit the file manually when there is a demand in invoking the lambda locally

**Comment by medikoo** [comment]
@maksym-opanasenko can you help finalizing this PR?

**Comment by jasonrowsell** [comment]
@medikoo Hey 👋 Since it's been over 30 days, I've created a PR to fix this, with the test required
https://github.com/serverless/serverless/pull/11629
@ me if any questions, happy to discuss

**Comment by medikoo** [comment]
Addressed with #11629

**Pull request closed without merging**